server/src-lib/Hasura/GraphQL/Execute/LiveQuery/Poll.hs

```diff
@@ -263,83 +289,164 @@ dumpPollerMap extended lqMap =
   fmap J.toJSON $ do
     entries <- STM.atomically $ ListT.toList $ STMMap.listT lqMap
     forM entries $ \(PollerKey role query, Poller cohortsMap ioState) -> do
-      PollerIOState threadId metrics <- STM.atomically $ STM.readTMVar ioState
-      metricsJ <- dumpRefetchMetrics metrics
+      PollerIOState threadId pollerId <- STM.atomically $ STM.readTMVar ioState
       cohortsJ <-
         if extended
         then Just <$> dumpCohortMap cohortsMap
         else return Nothing
       return $ J.object
         [ "role" J..= role
         , "thread_id" J..= show (Immortal.threadId threadId)
+        , "poller_id" J..= pollerId
         , "multiplexed_query" J..= query
         , "cohorts" J..= cohortsJ
-        , "metrics" J..= metricsJ
-        ]
-  where
-    dumpRefetchMetrics metrics = do
-      snapshotS <- Metrics.read $ _rmSnapshot metrics
-      queryS <- Metrics.read $ _rmQuery metrics
-      pushS <- Metrics.read $ _rmPush metrics
-      totalS <- Metrics.read $ _rmTotal metrics
-      return $ J.object
-        [ "snapshot" J..= dumpStats snapshotS
-        , "query" J..= dumpStats queryS
-        , "push" J..= dumpStats pushS
-        , "total" J..= dumpStats totalS
         ]
 
-    dumpStats stats =
-      J.object
-      [ "mean" J..= Metrics.mean stats
-      , "variance" J..= Metrics.variance stats
-      , "count" J..= Metrics.count stats
-      , "min" J..= Metrics.min stats
-      , "max" J..= Metrics.max stats
-      ]
+-- | An ID to track unique 'Poller's, so that we can gather metrics about each
+-- poller
+newtype PollerId = PollerId { unPollerId :: UUID.UUID }
+  deriving (Show, Eq, Generic, J.ToJSON)
+
+-- | Execution information related to a cohort on a poll cycle
+data CohortExecutionDetail
```

Comment:
Incredibly minor nitpick, but this type seems like it should be named `CohortExecutionDetails`, plural (and the same goes for `BatchExecutionDetail` and `PollDetail` below). It contains multiple details, not one detail!